lib/routes/ehentai/ehapi.js

```diff
@@ -23,26 +25,53 @@ function ehgot(url) {
     }
 }
 
-async function parsePage(cache, data, get_bittorrent = false) {
+function ehgot_thumb(cache, thumb_url) {
+    return cache.tryGet(thumb_url, async () => {
+        try {
+            const buffer = await got({ method: 'get', url: thumb_url, headers });
+            const data = new Buffer.from(buffer.rawBody).toString('base64');
+            const ext = path.extname(thumb_url).slice(1);
+            return `data:image/${ext};base64,${data}`;
+        } catch (e) {
+            logger.warn('Cannot download thumbnail: ' + thumb_url, e);
+            return thumb_url;
+        }
+    });
+}
+
+async function parsePage(cache, data, get_bittorrent = false, embed_thumb = false) {
     const $ = cheerio.load(data);
-    const table = $('table[class="itg gltc"] tbody');
+    const table = $('table[class^="itg glt"] tbody');
     if (!table) {
         return [];
     }
 
     async function parseTableRow(cache, element) {
         const el = $(element);
-        const el_a = el.find('td[class="gl3c glname"] a');
-        const el_img = el.find('td.gl2c div.glthumb div img');
-        const title = el_a.find('div.glink').html();
-        const thumbnail = el_img.attr('data-src') ? el_img.attr('data-src') : el_img.attr('src');
-        const description = `<img src='${thumbnail}' alt='thumbnail'>`;
+        const title = el.find('div.glink').html();
         const pubDate = el.find('div[id^="posted_"]').html();
+        // match "Minimal", "Minimal+" and "Compact"
+        let el_a = el.find('td[class^="gl3"] a');
+        let el_img = el.find('td[class^=gl2] div.glthumb div img');
+        // if not, match "Extended"
+        if (el_a && el_img) {
+            el_a = el.find('td[class^="gl1"] a');
+            el_img = el_a.find('img');
+        }
```

Comment:
Double-check your condition:

In Minimal listing,
```js
el_a = el.find('td[class^="gl3"] a');
el_img = el.find('td[class^=gl2] div.glthumb div img')
```

`el_a && el_img` will be `true` then
```js
el_a = el.find('td[class^="gl1"] a');
el_img = el_a.find('img');
```
which only appears in "Extended" listing